Apps/WebClient/src/ClientApp/src/stores/delegate.ts

```diff
@@ -0,0 +1,125 @@
+import { defineStore } from "pinia";
+import { ref } from "vue";
+
+import { DataSource } from "@/constants/dataSource";
+import { DelegateInvitationStatus } from "@/constants/delegateInvitationStatus";
+import { ErrorSourceType, ErrorType } from "@/constants/errorType";
+import { container } from "@/ioc/container";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import { ResultError } from "@/models/errors";
+import { CreateDelegateInvitationRequest } from "@/models/sharing/createDelegateInvitationRequest";
+import { DelegateInvitation } from "@/models/sharing/delegateInvitation";
+import { IDelegateService, ILogger } from "@/services/interfaces";
+import { useErrorStore } from "@/stores/error";
+
+export enum DelegateInvitationDialogStep {
+    contact = 0,
+    nickname = 1,
+    dataSources = 2,
+    expiryDate = 3,
+    review = 4,
+    sharingCode = 5,
+}
+
+export interface DelegateInvitationDialogState {
+    mode: "Create" | "Modify";
+    step: DelegateInvitationDialogStep;
+    nickname?: string;
+    email?: string;
+    expiryDate?: string;
+    dataSources?: DataSource[];
+}
+
+const defaultInvitationDialogState: DelegateInvitationDialogState = {
+    mode: "Create",
+    step: DelegateInvitationDialogStep.contact,
+};
+
+export const useDelegateStore = defineStore("delegate", () => {
+    const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    const delegateService = container.get<IDelegateService>(
+        SERVICE_IDENTIFIER.DelegateService
+    );
+
+    const errorStore = useErrorStore();
+
+    const invitations = ref<DelegateInvitation[]>([]);
+    const invitationDialogState = ref<DelegateInvitationDialogState>({
+        ...defaultInvitationDialogState,
+    });
+    const invitationsAreLoading = ref(false);
+    const error = ref<ResultError>();
+    const statusMessage = ref("");
+
+    function setInvitationsError(incomingError: ResultError) {
+        error.value = incomingError;
+        statusMessage.value = incomingError.resultMessage;
+        invitationsAreLoading.value = false;
+    }
+
+    function submitInvitationDialog(): Promise<string | void> {
+        if (invitationDialogState.value.mode === "Create") {
+            return createInvitation({
+                status: DelegateInvitationStatus.Pending,
+                nickname: invitationDialogState.value.nickname,
+                email: invitationDialogState.value.email,
+                expiryDate: invitationDialogState.value.expiryDate,
+                dataSources: invitationDialogState.value.dataSources,
+            });
+        }
+        // TODO: Remove this once all paths are covered
+        return Promise.resolve();
+    }
```

Comment:
It will be easiest if each page of the dialog has a separate store action to submit (since they will have different parameters).  This one is set up nicely for the final "review" page, but the naming implies there is only one way to submit the dialog and there won't be other submit actions.

I'd also suggest we may want to throw an error for states that are unsupported instead of returning a successful promise.

Reply:
I'll continue to develop this as I implement the front end.